recipes/imagl/all/conanfile.py

```diff
@@ -14,8 +13,8 @@ class ImaglConan(ConanFile):
     description = "A lightweight library to load image for OpenGL application."
     topics = ("opengl", "texture", "image")
     settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "fPIC": [True, False], "with_png": [True, False]}
-    default_options = {"shared": False, "fPIC": True, "with_png": True}
+    options = {"shared": [True, False], "fPIC": [True, False], "with_png": [True, False], "with_jpeg": [True, False], "allow_clang_11": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "with_png": True, "with_jpeg": True, "allow_clang_11": False}
```

Comment:
Putting all options on separate lines is much more readable.
The recipes should be optimized for maintainability instead of compactness. 
```suggestion
    options = {
        "shared": [True, False],
        "fPIC": [True, False],
        "with_png": [True, False],
        "with_jpeg": [True, False],
        "allow_clang_11": [True, False]
    }
    default_options = {
        "shared": False,
        "fPIC": True,
        "with_png": True,
        "with_jpeg": True,
        "allow_clang_11": False,
    }
```

Reply:
The option `allow_clang_11` is questionable and should imho be removed and handled differently.
Can't this option be just removed?

Reply:
CCI docker images do not correctly install the right version of `libc++` so it fails to build

Reply:
So doesn't that mean that adding a `allow_clang_11` option isn't an actual fix?

Reply:
Yes, it is absolutely not an actual fix. But this workaround is needed to pass CCI builds. I could remove it when CCI will have the right libc++ version in clang 11 docker image.
I reported the issue conan-io/conan-docker-tools#251 and proposed a PR for this: conan-io/conan-docker-tools#252, but it's not applied yet.

Reply:
What do you think about throwing a `ConanInvalidConfiguration` + a FIXME + some explanation why and come back to this later?

Reply:
If we do this, there will be no way to let user build imaGL with `--build=missing` conan option when using clang 11. That the reason why I added this bad looking `allow_clang_11` option.

Reply:
It would currently be impossible to build clang 11, but it will become possible when the problem in the docker images is fixed. Then, the `Error` can be removed.

I want to avoid adding this option because adding an option is easy, but removing it is hard.
Alternatively, you can create a tristate `allow_clang_11` option, with values: `[None, True, False]` and default `None`. When configured value `!= None`, print a warning about the option being unstable and prone to removal in the future.